subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/JvmGradlePluginProjectInitDescriptor.java

```diff
@@ -59,25 +59,33 @@ public void generateProjectBuildScript(String projectName, InitSettings settings
             g.propertyAssignment(null, "implementationClass", withPackage(settings, pluginClassName), true);
         }));
 
-        BuildScriptBuilder.Expression functionalTestSourceSet = buildScriptBuilder.createContainerElement("Add a source set for the functional test suite", "sourceSets", "functionalTest", "functionalTestSourceSet");
-        buildScriptBuilder.methodInvocation(null, "gradlePlugin.testSourceSets", functionalTestSourceSet);
-
-        BuildScriptBuilder.Expression functionalTestConfiguration = buildScriptBuilder.containerElementExpression("configurations", "functionalTestImplementation");
-        BuildScriptBuilder.Expression testConfiguration = buildScriptBuilder.containerElementExpression("configurations", "testImplementation");
-        buildScriptBuilder.methodInvocation(null, functionalTestConfiguration, "extendsFrom", testConfiguration);
-        BuildScriptBuilder.Expression functionalTest = buildScriptBuilder.taskRegistration("Add a task to run the functional tests", "functionalTest", "Test", b -> {
-            b.propertyAssignment(null, "testClassesDirs", buildScriptBuilder.propertyExpression(functionalTestSourceSet, "output.classesDirs"), true);
-            b.propertyAssignment(null, "classpath", buildScriptBuilder.propertyExpression(functionalTestSourceSet, "runtimeClasspath"), true);
-            if(getTestFrameworks().contains(BuildInitTestFramework.SPOCK) || getTestFrameworks().contains(BuildInitTestFramework.JUNIT_JUPITER)) {
-                b.methodInvocation(null, "useJUnitPlatform");
-            }
-        });
-        buildScriptBuilder.taskMethodInvocation("Run the functional tests as part of `check`", "check", "Task", "dependsOn", functionalTest);
-        if(getTestFrameworks().contains(BuildInitTestFramework.SPOCK) || getTestFrameworks().contains(BuildInitTestFramework.JUNIT_JUPITER)) {
-            if (!buildScriptBuilder.isUsingTestSuites()) {
+        final BuildScriptBuilder.Expression functionalTestSourceSet;
```

Comment:
Something isn't quite right with the build script. I'm seeing a lot of duplication:
```

dependencies {
    // Use the awesome Spock testing and specification framework
    testImplementation 'org.spockframework:spock-core:2.0-groovy-3.0'
}

testing {
    suites {
        test(JvmTestSuite) {
            useSpock()

            dependencies {
                implementation 'org.spockframework:spock-core:2.0-groovy-3.0'
            }
        }

        functionalTest(JvmTestSuite) {
            useSpock()

            dependencies {
                implementation project
                implementation gradleTestKit()
                implementation gradleApi()
            }

            targets {
                all {
                    testTask.configure { shouldRunAfter(test) } 
                }
            }
        }
    }
}
```

I think this should look more like:
```
testing {
    suites {
        test(JvmTestSuite) {
            useSpock()
        }

        functionalTest(JvmTestSuite) {
            useSpock()

            targets {
                all {
                    testTask.configure { shouldRunAfter(test) } 
                }
            }
        }
    }
}
```